client/src/components/form/form.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useContext } from 'react';
+
+const FormHeader: React.FC<React.HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
+  return (
+    <h2 {...props} >{children}</h2>
+  )
+}
+
+const FormLabel: React.FC<React.HTMLAttributes<HTMLSpanElement>> = ({ children, ...props }) => {
+  return (
+    <span {...props} >{children}</span>
+  )
+}
+
+const FormInput: React.FC<React.HTMLAttributes<HTMLElement>> = (props) => {
+  return (
+    <input {...props} />
+  )
+}
+
+const FormBody: React.FC<React.FormHTMLAttributes<HTMLFormElement>> = ({ children, ...props }) => {
+  return (
+    <form {...props} >{children}</form>
+  )
+}
+
+const FormButton: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({ children, ...props }) => {
+  return (
+    <button {...props} >{children}</button>
+  )
+}
+
+const FormClose: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({ children, ...props }) => {
+  return (
+    <FormContext.Consumer>
+      {formContext => {
+        return (
+          <button {...props} onClick={formContext.toggleShow}> {children} </button>
+        )
+      }}
+    </FormContext.Consumer>
+  )
+}
+
+const initialState = {
+  show: true,
+  toggleShow: () => {},
+};
+
+const FormContext = React.createContext(initialState);
+
+function Form(props: any){
+  const [show, setShow] = useState(true);
+  const toggleShow = () => {
+    setShow(!show);
```

Comment:
The `Form` component hiding itself may not work out in the long term. What does it mean for a form to `hide`? What takes it's place? How does another component interact with this component?